From c641a5dc32f82d1ffdb0637e4a414fe6d46551c1 Mon Sep 17 00:00:00 2001
From: Noam Lessner <nlessner@habana.ai>
Date: Tue, 18 Feb 2020 15:01:12 +0200
Subject: [PATCH] [SW-8229] goya2: arcboot_ram header offset

Execute arcboot_ram at offset of header when copying from
flash that includes size and version. Two double words offset
the start of code to address 0x8

Change-Id: I984461e2c414e0a7f40c6c727ae2b4233d88a8be
Signed-off-by: Noam Lessner <nlessner@habana.ai>
---

diff --git a/app/arcboot_rom.c b/app/arcboot_rom.c
index fdb132e..2f11680 100644
--- a/app/arcboot_rom.c
+++ b/app/arcboot_rom.c
@@ -347,7 +347,7 @@
 	arcboot_rom_status_update(PROGRESS_ARCBOOT_RAM_START);
 
 	/* jump to iccm start location */
-	goto *(void *)(0);
+	goto *(void *)(ARCBOOT_RAM_HEADER);
 }
 
 int arcboot_rom_main(void)
diff --git a/include/arcboot.h b/include/arcboot.h
index e6049d3..962d3f4 100644
--- a/include/arcboot.h
+++ b/include/arcboot.h
@@ -49,6 +49,12 @@
 #define PPBOOT_IMAGE_CODE_ALIGN			4
 #define PPBOOT_IMAGE_CODE_OFFSET		(PPBOOT_IMAGE_OFFSET + \
 						 PPBOOT_IMAGE_CODE_ALIGN)
+
+#define IMAGE_SIZE_LEN				sizeof(uint32_t)
+#define IMAGE_VERSION_LEN			sizeof(uint32_t)
+#define ARCBOOT_RAM_HEADER			(IMAGE_SIZE_LEN + \
+						 IMAGE_VERSION_LEN)
+
 #define COLD_RST_FLOPS_RECOVERY_FLAG_SHIFT	0x0
 #define COLD_RST_FLOPS_RECOVERY_FLAG_MASK	0x1
 #define COLD_RST_FLOPS_VALIDATION_PASS_SHIFT	0x1
